### ISSUE TITLE
Unable to connect to Server 2019 to manage tasks

### ISSUE THREAD

**Issue by rasitha1** [dahall/TaskScheduler#819]
**Describe the bug**

We have been using this library to manage scheduled tasks and has worked great. Recently with Server 2019 servers getting into the mix, this library can no longer connect. 

```
System.Runtime.InteropServices.COMException (0x80070032): The request is not supported. (Exception from HRESULT: 0x80070032)
   at Microsoft.Win32.TaskScheduler.V2Interop.ITaskService.Connect(Object serverName, Object user, Object domain, Object password)
   at Microsoft.Win32.TaskScheduler.TaskService.Connect()
   at Microsoft.Win32.TaskScheduler.TaskService..ctor(String targetServer, String userName, String accountDomain, String password, Boolean forceV1)
   at TaskSchedulerTestApp.Program.Main(String[] args)
```

**To Reproduce**
1. Run the following sample console app against a 2019 Server

**Expected behavior**
Should list tasks on the remote server. (Works for 2016 and below but not on 2019)

**Environment (please complete the following information):**
 - Host Windows version (the one running the code): Windows 10
 - Library version: 2.8.14
 - .Net Framework: 4.7.2
 - Remote Windows version (if applicable): Windows Server 2019 Datacenter v10.0 (17763)

**Additional context**

Here's the sample app code:

```
namespace TaskSchedulerTestApp
{
    class Program
    {
        static void Main(string[] args)
        {
            try
            {
                var server = "my2019server";
                var user = "admin_user";
                var domain = "mydomain";
                var pass = "secret";

                var taskService = new TaskService(server, user, domain, pass);

                Debug.Assert(taskService.Connected, "Not connected");

                foreach (var task in taskService.AllTasks)
                {
                    Console.WriteLine(task.Name);
                }

                Console.WriteLine("Done");
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex);
            }
            Console.ReadLine();
        }
    }
}
```
Do you know if this is a known case? Or is there a different way to do this now?

**Comment by dahall** [comment]
This is new information. Have you tried running the [troubleshooting tool](https://github.com/dahall/TaskSchedulerConfig) on both the client and the server to see if there are system configuration options causing the problem?

**Comment by rasitha1** [comment]
Ran the diagnostics tool and there are two entries:

V2 Remote Access: "Remote Task Management" rule on the firewall is not enabled
V2 Remote Access: "Remote Registry" service is not running

However, when I run it against a 2016 Server that TaskService can successfully connect and pull data, I still get the same two items. So I'm guessing that's not a real issue?

**Comment by dahall** [comment]
It is a real problem. You shouldn't be able to connect without those two services running. Are you constructing the `TaskService` instance using the `forceV1` parameter set to `true`?

**Comment by rasitha1** [comment]
I'm not. (See my sample code above). 

Earlier I did try with forceV1=true and that let me connect to the server but none of the tasks would show up. The task list is empty. Also if I try to go to a sub folder it throws this error:

`System.ArgumentException: Path not found
Parameter name: path
   at Microsoft.Win32.TaskScheduler.TaskFolderCollection.get_Item(String path)`

If V2 Remote Access issues are there on Server 2016 how am I able to connect and manage tasks successfully on that server? Is it using a different protocol/technique when connecting to 2016?

I can try enabling those services and see but just from the comparison those are non issues for at least how we are using the library.

**Comment by dahall** [comment]
I have never been able to manage remote connections without at least the Remote Task Management set so I'm not sure how your WS2016 server is connecting. V1 connections are legacy and don't support new features like modern tasks and folders. However, they are a lot more forgiving/insecure when it comes to remote access. But again, with V2, Remote Task Management has to be available in my experience.

**Comment by dahall** [comment]
I'm installing WS2019 1903 as we speak. Hopefully I'll be able to test shortly.

**Comment by dahall** [comment]
Hm. It appears there is only a Server Core version of 1903. Does your 1809 install have the GUI (Desktop Experience)?

**Comment by rasitha1** [comment]
Yes, I can RDP into the box. It's a Windows Server 2019 Datacenter version.

**Comment by rasitha1** [comment]
For the 2 issues from Troubleshoot app, I was able to start the Remote Registry service but there's no existing firewall rule in either 2016 or 2019 server called "Remote Task Management". (The tool couldn't fix it either)

**Comment by dahall** [comment]
Ok. Once I get this installed I'll review what's in troubleshooter. I haven't looked at it for a while.

**Comment by dahall** [comment]
I'm still trying to find GUI WS19 bits. In the mean time, are you able to connect to this remote WS19 server using the Task Scheduler admin tool included with Windows?

**Comment by rasitha1** [comment]
Yes, at least it doesn't give an error when I point to the remote server. It shows the same "Remote Registry" service as not running. However it's not detecting the firewall rule being a problem when connecting remotely.

**Comment by dahall** [comment]
So I finally got the server running. I could not use the system tool or the library to connect to the remote WS19 box until I went into the "Allow an app through Windows Firewall" (in Control Panel) and enabled Domain and/or Private access to "Remote Event Log Management" and "Remote Scheduled Tasks Management".

Once connected however, I did find some incompatibilities when trying to retrieve certain tasks. I'm assuming from past experience that Microsoft has added new undocumented properties to the native COM objects. I will try to get to the bottom of those and will post an update to the library.

However, connecting is not a problem with those two firewall options enabled.
